src/topics/tools/lda_topics.cpp

```diff
@@ -1,76 +1,59 @@
 /**
  * @file lda_topics.cpp
  * @author Chase Geigle
+ * @author Matt Kelly
  */
 
 #include <fstream>
 #include <iostream>
 #include <string>
 #include <vector>
 
+#include "meta/caching/no_evict_cache.h"
 #include "meta/index/forward_index.h"
-#include "meta/util/fixed_heap.h"
+#include "meta/logging/logger.h"
+#include "meta/topics/topic_model.h"
 
 using namespace meta;
+using namespace meta::topics;
 
-int print_usage(const std::string& name)
+int print_topics(std::shared_ptr<index::cached_index<index::forward_index,
+                                                     caching::no_evict_cache>>
+                     idx,
+                 topic_model tm)
 {
-    std::cout
-        << "Usage: " << name
-        << " config_file model.phi num_words \n"
-           "\tPrints the top num_words words in each topic in the given model"
-        << std::endl;
-    return 1;
-}
-
-int print_topics(const std::string& config_file, const std::string& filename,
-                 size_t num_words)
-{
-    auto config = cpptoml::parse_file(config_file);
-    auto idx = index::make_index<index::forward_index>(*config);
-
-    std::ifstream file{filename};
-    while (file)
+    auto num_topics = tm.num_topics();
+    for (term_id i{0}; i < num_topics; ++i)
     {
-        std::string line;
-        std::getline(file, line);
-        if (line.length() == 0)
-            continue;
-        std::stringstream stream(line);
-        size_t topic;
-        stream >> topic;
-        std::cout << "Topic " << topic << ":" << std::endl;
-        std::cout << "-----------------------" << std::endl;
-
-        using scored_term = std::pair<term_id, double>;
-        auto pairs = util::make_fixed_heap<scored_term>(
-            num_words, [](const scored_term& a, const scored_term& b) {
-                return a.second > b.second;
-            });
+        std::cout << "Topic " << i << ":" << std::endl;
+        std::cout << "-----------------" << std::endl;
 
-        while (stream)
+        auto top_k = tm.top_k(i, 10);
+        for (auto& i : top_k)
         {
-            std::string to_split;
-            stream >> to_split;
-            if (to_split.length() == 0)
-                continue;
-            size_t idx = to_split.find_first_of(':');
-            term_id term{std::stoul(to_split.substr(0, idx))};
-            double prob = std::stod(to_split.substr(idx + 1));
-            pairs.emplace(term, prob);
+            std::cout << idx->term_text(term_id{i.tid}) << " (" << i.tid
```

Comment:
If you chance the definition of `tid` in `term_prob`, you can get rid of the cast here.